counter/api.py

```diff
@@ -64,15 +73,72 @@ def search_counter(self, filters: Query[CounterFilterSchema]):
 class ProductController(ControllerBase):
     @route.get(
         "/search",
-        response=PaginatedResponseSchema[ProductSchema],
+        response=PaginatedResponseSchema[SimpleProductSchema],
         permissions=[CanAccessLookup],
     )
     @paginate(PageNumberPaginationExtra, page_size=50)
-    def search_products(self, search: Annotated[str, MinLen(1)]):
-        return (
-            Product.objects.filter(
-                Q(name__icontains=search) | Q(code__icontains=search)
+    def search_products(self, filters: Query[ProductFilterSchema]):
+        return filters.filter(
+            Product.objects.order_by(
+                F("product_type__order").asc(nulls_last=True),
+                "product_type",
+                "name",
+            ).values()
+        )
+
+    @route.get(
+        "/search/detailed",
+        response=PaginatedResponseSchema[ProductSchema],
+        permissions=[IsCounterAdmin],
+        url_name="search_products_detailed",
+    )
+    @paginate(PageNumberPaginationExtra, page_size=50)
+    def search_products_detailed(self, filters: Query[ProductFilterSchema]):
+        """Get the detailed information about the products."""
+        return filters.filter(
+            Product.objects.select_related("club")
+            .prefetch_related("buying_groups")
+            .select_related("product_type")
+            .order_by(
+                F("product_type__order").asc(nulls_last=True),
+                "product_type",
+                "name",
             )
-            .filter(archived=False)
-            .values()
         )
+
+
+@api_controller("/product-type", permissions=[IsCounterAdmin])
+class ProductTypeController(ControllerBase):
+    @route.get("", response=list[ProductTypeSchema], url_name="fetch-product-types")
```

Comment:
```suggestion
    @route.get("", response=list[ProductTypeSchema], url_name="fetch_product_types")
```
Pour rester cohérent avec le reste, 99% des noms d'urls sont en snake case

Reply:
Ce serait pas bien de faire une réponse paginée ?